### PR TITLE
passwdqc: init at 2.0.3

### PR DESCRIPTION
**Pull request by simsor**
## Description of changes

passwdqc is a collection of tools to test password strength, generate strong passwords, and enforce them using PAM. It is hosted on OpenWall: https://www.openwall.com/passwdqc/

The package itself does not have any tests to run. Once built, I was able to successfully use `pwqgen` to generate a password.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by simsor** [comment]
Result of `nixpkgs-review pr 260903` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>passwdqc</li>
  </ul>
</details>

**Review by h7x4 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/pa/passwdqc/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+stdenv.mkDerivation rec {
+  pname = "passwdqc";
+  version = "2.0.3";
+
+  src = fetchzip {
+    url = "https://www.openwall.com/passwdqc/passwdqc-${version}.tar.gz";
+    hash = "sha256-EgPeccqS+DDDMBVMc4bd70EMnXFuyglftxuqoaYHwNY=";
+  };
+
+  propagatedBuildInputs = [
```
[the remark this answers is not in the document]
Comment (simsor) [reply pkgs/by-name/pa/passwdqc/package.nix:16]: Indeed, I misunderstood what `propagatedBuildInput` was doing

File: pkgs/by-name/pa/passwdqc/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "DEVEL_LIBDIR=/lib"
+    "INCLUDEDIR=/include"
+  ];
+
+  meta = with lib; {
+    description = "password/passphrase strength checking and enforcement";
+    homepage = "https://www.openwall.com/passwdqc/";
+    license = licenses.bsd0;
+    maintainers = [ maintainers.simsor ];
+  };
```
[the remark this answers is not in the document]
Comment (simsor) [reply pkgs/by-name/pa/passwdqc/package.nix:33]: The [PLATFORMS](https://www.openwall.com/passwdqc/PLATFORMS) file does not list macOS as a supported target. I will add a `platforms` field targeting Linux only

File: pkgs/by-name/pa/passwdqc/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    hash = "sha256-EgPeccqS+DDDMBVMc4bd70EMnXFuyglftxuqoaYHwNY=";
+  };
+
+  buildInputs = [
+    linux-pam
+    libxcrypt
+  ];
+
+  makeFlags = [
+    "DESTDIR=$(out)"
```
Comment (AndersonTorres) [inline pkgs/by-name/pa/passwdqc/package.nix:22]: Do not use DESTDIR on Nixpkgs.
```suggestion
```

File: pkgs/by-name/pa/passwdqc/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "BINDIR=/bin"
+    "DEVEL_LIBDIR=/lib"
+    "INCLUDEDIR=/include"
```
Comment (AndersonTorres) [inline pkgs/by-name/pa/passwdqc/package.nix:25]:
```suggestion
    "BINDIR=${placeholder "out"}/bin"
    "DEVEL_LIBDIR=${placeholder "out"}/lib"
    "INCLUDEDIR=${placeholder "out"}/include"
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/pa/passwdqc/package.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    "MANDIR=${placeholder "out"}/share/man"
+    "LOCALEDIR=${placeholder "out"}/share/locale"
+    "CONFDIR=${placeholder "out"}/etc"
+  ];
+
+  meta = with lib; {
+    description = "Password/passphrase strength checking and enforcement";
+    homepage = "https://www.openwall.com/passwdqc/";
+    platforms = platforms.linux;
+    license = licenses.bsd0;
```
Comment (h7x4) [inline pkgs/by-name/pa/passwdqc/package.nix:36]: This doesn't seem entirely correct

https://www.openwall.com/passwdqc/LICENSE

Parts of the code seems to be `bsd3` (and gpl2, but that's not included in the tarball?)

```suggestion
    license = with licenses; [ bsd0 bsd3 ];
```

**Pull request closed without merging**